controllers/storagecluster/cephcluster.go

```diff
@@ -1707,3 +1718,52 @@ func getFullRatio(specifiedRatio *float64) *float64 {
 	defaultRatio := 0.85
 	return &defaultRatio
 }
+
+// updateRookCephOperatorHostNetwork updates the rook-ceph-operator CSV to enable host network
+func updateRookCephOperatorHostNetwork(ctx context.Context, c client.Client, opNamespace string) error {
+	csvList := &opv1a1.ClusterServiceVersionList{}
+	labelSelector := labels.SelectorFromSet(map[string]string{
+		fmt.Sprintf("operators.coreos.com/rook-ceph-operator.%v", opNamespace): "",
+	})
+	listOpts := &client.ListOptions{
+		Namespace:     opNamespace,
+		LabelSelector: labelSelector,
+	}
+
+	// List the ClusterServiceVersion for rook-ceph-operator
+	err := c.List(ctx, csvList, listOpts)
+	if err != nil {
+		return err
+	}
+
+	// Check if we found exactly one ClusterServiceVersion for rook-ceph-operator
+	if len(csvList.Items) == 0 {
+		return fmt.Errorf("no ClusterServiceVersion found for rook-ceph-operator in namespace %s", opNamespace)
+	}
+	if len(csvList.Items) > 1 {
+		return fmt.Errorf("multiple ClusterServiceVersions found for rook-ceph-operator, skipping update during upgrade")
+	}
+	rookCSV := &csvList.Items[0]
+
+	// Update the rook-ceph-operator deployment in CSV to enable host network
+	updated := false
+	for i := range rookCSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+		deployment := &rookCSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[i]
+		if deployment.Name == "rook-ceph-operator" {
+			// Enable host network
+			deployment.Spec.Template.Spec.HostNetwork = true
```

Comment:
Why are we using this approach of getting the CSV & patching the deployment spec inside it, For setting values from ocs-operator to rook, we already have the ocs-operator-config CM,
You can add the value based on your conditions to that CM & that CM's values can be fed to rook operator via CSV env vars. See example here https://github.com/red-hat-storage/ocs-operator/pull/2975